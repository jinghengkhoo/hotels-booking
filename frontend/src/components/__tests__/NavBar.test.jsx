// import React from 'react';
// import { render, screen, fireEvent } from '@testing-library/react';
// import '@testing-library/jest-dom';
// import { BrowserRouter as Router } from 'react-router-dom';
// import NavBar from '../NavBar';

// describe('NavBar', () => {
//     test('renders the menu button', () => {
//         render(
//             <Router>
//                 <NavBar textColor="white" currency="USD" />
//             </Router>
//         );
//         const menuButton = screen.getByText('MENU');
//         expect(menuButton).toBeInTheDocument();
//     });

//     // test('renders the logo button', () => {
//     //     render(
//     //         <Router>
//     //             <NavBar textColor="white" currency="USD" />
//     //         </Router>
//     //     );
//     //     const logoButton = screen.getByText('Travelust');
//     //     expect(logoButton).toBeInTheDocument();
//     // });

//     // test('renders the currency dropdown', () => {
//     //     render(
//     //         <Router>
//     //             <NavBar textColor="white" currency="USD" />
//     //         </Router>
//     //     );
//     //     const currencyDropdown = screen.getByText('USD');
//     //     expect(currencyDropdown).toBeInTheDocument();
//     // });

//     // test('opens the currency dropdown when clicked', () => {
//     //     render(
//     //         <Router>
//     //             <NavBar textColor="white" currency="USD" />
//     //         </Router>
//     //     );
//     //     const currencyDropdown = screen.getByText('USD');
//     //     fireEvent.click(currencyDropdown);
//     //     const currencyOption = screen.getByText('SGD');
//     //     expect(currencyOption).toBeInTheDocument();
//     // });

//     // test('renders the user icon', () => {
//     //     render(
//     //         <Router>
//     //             <NavBar textColor="white" currency="USD" />
//     //         </Router>
//     //     );
//     //     const userIcon = screen.getByTestId('user-icon');
//     //     expect(userIcon).toBeInTheDocument();
//     // });

//     // test('renders the login button when user is not logged in', () => {
//     //     render(
//     //         <Router>
//     //             <NavBar textColor="white" currency="USD" />
//     //         </Router>
//     //     );
//     //     const loginButton = screen.getByText('Login/Register');
//     //     expect(loginButton).toBeInTheDocument();
//     // });

//     // test('renders the user email and logout button when user is logged in', () => {
//     //     render(
//     //         <Router>
//     //             <NavBar textColor="white" currency="USD" />
//     //         </Router>
//     //     );
//     //     const user = { email: 'test@example.com' };
//     //     const logoutButton = screen.getByText('Logout');
//     //     expect(logoutButton).toBeInTheDocument();
//     // });
// });