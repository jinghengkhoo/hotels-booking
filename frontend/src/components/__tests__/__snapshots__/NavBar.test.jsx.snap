// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<div
  className="navbar bg-transparent z-20 py-4 text-white"
>
  <div
    className="flex justify-left items-center w-1/4 pl-16 relative"
  >
    <button
      className="btn btn-ghost"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars mr-2"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64l384 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 128C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32l384 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 288c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32L32 448c-17.7 0-32-14.3-32-32s14.3-32 32-32l384 0c17.7 0 32 14.3 32 32z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span
        className="font-normal"
      >
        MENU
      </span>
    </button>
  </div>
  <div
    className="container mx-auto flex justify-center w-1/2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn btn-ghost text-center font-bold text-xl"
      >
        Travelust
      </button>
    </a>
  </div>
  <div
    className="flex justify-end items-center space-x-4 pr-4 w-1/4"
  >
    <div
      className="dropdown dropdown-end justify-end"
      onMouseLeave={[Function]}
    >
      <div
        className="btn btn-ghost rounded-btn"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        USD
      </div>
    </div>
    <div
      className="flex items-center"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user mr-2"
        data-icon="user"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512l388.6 0c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304l-91.4 0z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <span>
        <div>
          <a
            href="/login"
            onClick={[Function]}
          >
            <button
              className="btn btn-ghost"
              id="login"
            >
              <span
                className="font-normal"
              >
                Login/Register
              </span>
            </button>
          </a>
        </div>
      </span>
    </div>
  </div>
</div>
`;
